### PR TITLE
Migrate PodLifeTime to plugin

### PR DESCRIPTION
**Pull request by a7i**

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -177,51 +179,54 @@ func runPodLifetimeStrategy(
...
-			nodes,
-			getPodsAssignedToNode,
-			false,
-			evictCritical,
-			false,
-			false,
-			evictions.WithPriorityThreshold(thresholdPriority),
-		),
 		getPodsAssignedToNode,
+		true,
```
Comment (ingvagabund) [inline test/e2e/e2e_test.go:206]: Has anything changed that requires `evictLocalStoragePods=true`?

**Review by ingvagabund — CHANGES_REQUESTED** [review]

File: pkg/descheduler/strategy_migration.go [hunk 2]
```diff
@@ -130,6 +131,40 @@ var pluginsMap = map[string]func(ctx context.Context, nodes []*v1.Node, params *
...
+		}
+		if podLifeTimeParams.States != nil {
+			states = append(states, podLifeTimeParams.States...)
+		}
+
+		args := &componentconfig.PodLifeTimeArgs{
+			Namespaces:            params.Namespaces,
+			LabelSelector:         params.LabelSelector,
+			MaxPodLifeTimeSeconds: podLifeTimeParams.MaxPodLifeTimeSeconds,
+			States:                states,
```
[the remark this answers is not in the document]
Comment (a7i) [reply pkg/descheduler/strategy_migration.go:152]: Is there a reason that it cannot be removed from `PodLifeTimeArgs` and continue to combine them [like this](https://github.com/kubernetes-sigs/descheduler/pull/913/files#diff-548663336f315a426f99e4cee346e72f48e31820fd5dba6fac5e4acd289a0506R140-R146)?

File: pkg/framework/plugins/podlifetime/pod_lifetime.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	}
+
+	// Should sort Pods so that the oldest can be evicted first
+	// in the event that PDB or settings such maxNoOfPodsToEvictPer* prevent too much eviction
+	podutil.SortPodsBasedOnAge(podsToEvict)
+
+	for _, pod := range podsToEvict {
+		d.handle.Evictor().Evict(ctx, pod, evictions.EvictOptions{})
+		if d.handle.Evictor().NodeLimitExceeded(nodeMap[pod.Spec.NodeName]) {
+			break
```
[the remark this answers is not in the document]
Comment (a7i) [reply pkg/framework/plugins/podlifetime/pod_lifetime.go:132]: Good catch. That makes sense

File: test/e2e/e2e_test.go (hunk 1 above)
Comment (a7i) [reply test/e2e/e2e_test.go:206]: No, just copy-pasta ©️

File: pkg/descheduler/strategy_migration.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ingvagabund) [reply pkg/descheduler/strategy_migration.go:152]: That's only a comment for anyone who would like to track this back in the history. It has no effect for the review itself.

**Pull request merged**